File: lib_common/src/main/java/com/example/common/base/binding/CommonBindingAdapter.java
package com.example.common.base.binding;

import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.databinding.BindingAdapter;

import com.example.common.imageloader.ImageLoader;

/**
 * Created by WangYanBin on 2020/6/10.
 * 基础全局xml绑定
 */
public class CommonBindingAdapter {

    //---------------------------------------------图片注入开始---------------------------------------------
    @BindingAdapter(value = {"app:imageRes"})
    public static void setImageRes(ImageView imageView, int imageRes) {
        imageView.setImageResource(imageRes);
    }

    @BindingAdapter(value = {"app:imageUrl"})
    public static void setImageDisplay(ImageView view, String url) {
        ImageLoader.Companion.getInstance().displayImage(view, url);
    }

    @BindingAdapter(value = {"app:imageRoundUrl", "app:imageRoundRadius"})
    public static void setImageRoundDisplay(ImageView view, String url, int roundingRadius) {
        ImageLoader.Companion.getInstance().displayRoundImage(view, url, roundingRadius);
    }

    @BindingAdapter(value = {"app:imageCircleUrl"})
    public static void setImageCircleDisplay(ImageView view, String url) {
        ImageLoader.Companion.getInstance().displayCircleImage(view, url);
    }
    //---------------------------------------------图片注入结束---------------------------------------------

    //---------------------------------------------系统方法注入开始---------------------------------------------
    @BindingAdapter(value = {"app:textColorRes"})
    public static void setTextColor(TextView textView, int textColorRes) {
        textView.setTextColor(textView.getResources().getColor(textColorRes));
    }

    @BindingAdapter(value = {"app:visible"})
    public static void setVisible(View view, boolean visible) {
        view.setVisibility(visible ? View.VISIBLE : View.GONE);
    }

    @BindingAdapter(value = {"app:selected"})
    public static void setSelected(View view, boolean select) {
        view.setSelected(select);
    }

    @BindingAdapter(value = {"adjustWidth"})
    public static void setAdjustWidth(View view, int adjustWidth) {
        ViewGroup.LayoutParams params = view.getLayoutParams();
        params.width = adjustWidth;
        view.setLayoutParams(params);
    }

    @BindingAdapter(value = {"app:adjustHeight"})
    public static void setAdjustHeight(View view, int adjustHeight) {
        ViewGroup.LayoutParams params = view.getLayoutParams();
        params.height = adjustHeight;
        view.setLayoutParams(params);
    }
    //---------------------------------------------系统方法注入结束---------------------------------------------

}

File: lib_common/src/main/java/com/example/common/base/binding/CommonBindingAdapter.kt
package com.example.common.base.binding

import android.annotation.SuppressLint
import android.widget.ImageView
import androidx.databinding.BindingAdapter
import com.chad.library.adapter.base.BaseQuickAdapter
import com.example.common.imageloader.ImageLoader
import com.example.common.widget.XWebView
import com.example.common.widget.xrecyclerview.XRecyclerView

/**
 * Created by WangYanBin on 2020/6/10.
 * 全局通用工具类
 * 复用性高的代码，统一放在common中
 * 比如列表页都需要设置适配器属性，富文本加载网页
 */
object CommonBindingAdapter {

    /**
     * 适配器
     */
    @JvmStatic
    @BindingAdapter(value = ["adapter"])
    fun <T : BaseQuickAdapter<*, *>> setAdapter(rec: XRecyclerView, adapter: T) {
        rec.setAdapter(adapter)
    }

    /**
     * 加载网页
     */
    @JvmStatic
    @SuppressLint("SetJavaScriptEnabled", "JavascriptInterface", "AddJavascriptInterface")
    @BindingAdapter(value = ["loadUrl"])
    fun setLoadUrl(webView: XWebView, loadPageUrl: String) {
        webView.loadUrl(loadPageUrl)
    }

    /**
     * 加载本地网页
     */
    @JvmStatic
    @SuppressLint("SetJavaScriptEnabled", "JavascriptInterface", "AddJavascriptInterface")
    @BindingAdapter(value = ["loadAssetUrl"])
    fun setLoadAssetsUrl(webView: XWebView, assetPath: String) {
        webView.loadUrl("file:///android_asset/$assetPath")
    }

    /**
     * 加载图片
     */
    @JvmStatic
    @BindingAdapter(value = ["display"])
    fun setDisplay(view: ImageView, url: String) {
        ImageLoader.instance.displayImage(view, url)
    }

    /**
     * 加载图片（比例缩放）
     */
    @JvmStatic
    @BindingAdapter(value = ["displayZoom"])
    fun setDisplayZoom(view: ImageView, url: String) {
        ImageLoader.instance.displayZoomImage(view, url)
    }

    /**
     * 加载图片（带圆角）
     */
    @JvmStatic
    @BindingAdapter(value = ["displayRound"])
    fun setDisplayRound(view: ImageView, url: String) {
        ImageLoader.instance.displayRoundImage(view, url, 5)
    }

    /**
     * 加载图片（圆形）
     */
    @JvmStatic
    @BindingAdapter(value = ["displayCircle"])
    fun setDisplayCircle(view: ImageView, url: String) {
        ImageLoader.instance.displayCircleImage(view, url)
    }

}